src/main/java/fruitshop/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,26 @@
+package fruitshop.strategy;
+
+import fruitshop.model.FruitTransaction;
+import fruitshop.model.Storage;
+
+public class SupplyOperation implements OperationHandler {
+    public static final String READ_FILE_PATH = "src/main/resources/reportToRead.csv";
+    private final Storage storage;
+
+    public SupplyOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handleTransaction(FruitTransaction fruitTransaction) {
+        int newQuantity = fruitTransaction.getQuantity()
+                + storage.getQuantity(fruitTransaction.getFruit());
+        if (newQuantity >= 0) {
+            storage.put(fruitTransaction.getFruit(), newQuantity);
+        } else {
+            throw new RuntimeException("negative balance " + newQuantity
+                    + " cannot be recorded at "
+                    + PurchaseOperation.class + " from fail " + READ_FILE_PATH);
```

Comment:
The exception message is not accurate. It refers to `PurchaseOperation.class` which is not related to the `SupplyOperation`. Also, the hardcoded file path `READ_FILE_PATH` should not be used in the exception message. Instead, provide a message that accurately reflects the error in the current context.